### PR TITLE
feat(Issue 75)

### PR DESCRIPTION
**Pull request by icamacho1**
Add support for Service Type in instances.

### CODE REVIEW

File: controllers/instance/instance_controller_test.go [hunk 1]
```diff
@@ -78,6 +80,12 @@ var _ = Describe("Reconcile", Label("controller"), func() {
 					},
 					Labels: labels,
 					Env:    labels,
+					Network: proxyv1alpha1.Network{
+						Service: proxyv1alpha1.ServiceSpec{
+							Enabled: true,
+							Type:    &lb,
```
Comment (snorwin) [inline controllers/instance/instance_controller_test.go:86]:
```suggestion
							Type:     ptr.To(corev1.ServiceTypeLoadBalancer),
```

File: apis/proxy/v1alpha1/instance_types.go [hunk 2]
```diff
@@ -100,6 +100,10 @@ type RouteSpec struct {
 type ServiceSpec struct {
 	// Enabled will toggle the creation of a Service.
 	Enabled bool `json:"enabled"`
+	// Type will define the service type, defaults to ClusterIP
+	// +optional
+	// +kubebuilder:default="corev1.ServiceTypeClusterIP"
+	Type *corev1.ServiceType `json:"type"`
```
Comment (snorwin) [inline apis/proxy/v1alpha1/instance_types.go:106]:
```suggestion
	Type *corev1.ServiceType `json:"type,omitempty"`
```

**Comment by icamacho1** [comment]
@snorwin 

Thanks for the comments 😄  I committed your suggestions.

**Review by m-terra — APPROVED** [review]
@icamacho1 thank you for your contribution
@snorwin thank you for your review

**Pull request merged**